rexfw/samplers/rwmc.py

```diff
@@ -13,20 +13,31 @@
 
 class RWMCSampler(AbstractSampler):
 
-    def __init__(self, pdf, state, stepsize, variable_name='x'):
+    def __init__(self, pdf, state, stepsize, timestep_adaption_limit=0,
+                 adaption_uprate=1.05, adaption_downrate=0.95,
+                 variable_name='x'):
 
         super(RWMCSampler, self).__init__(pdf, state, variable_name)
-        
+
         self.stepsize = stepsize
+        self.timestep_adaption_limit = timestep_adaption_limit
+        self.adaption_uprate = adaption_uprate
+        self.adaption_downrate = adaption_downrate
         self._last_move_accepted = False
         self._n_moves = 0
 
     @property
     def last_draw_stats(self):
-        
+
         return {self.variable_name: RWMCSampleStats(self._last_move_accepted, 
                                                     self._n_moves, self.stepsize)}
 
+    def _adapt_stepsize(self):
+        if self._last_move_accepted:
+            self.stepsize *= self.adaption_uprate
```

Comment:
Only thing I can think of here is that if the stepsize is bounded (e.g. a std deviation) this could potentially adapt the step size to a unsupported value.

Reply:
Hmm actually don't think so - if the stepsize is > 0 in the beginning, it might get lower and lower, but will never go below zero (unless floating point precision kicks in, but it's probably going to fail before). And a higher stepsize is not a problem.
Currently, the random walk Metropolis-Hastings implementation just proposes states uniformly from an interval [x - stepsize, x + stepsize], so there shouldn't be an issue either. Same for HMC: negative stepsize doesn't make sense, but this is not going to happen, as we're always multiplying a positive number by a positive number.

Reply:
Ah gotcha, I had read that as a decrement. Makes sense to me :+1: